nav2_simple_commander/launch/gz_security_demo_launch.py

```diff
@@ -0,0 +1,206 @@
+
+    # Map fully qualified names to relative ones so the node's namespace can be prepended.
+    # In case of the transforms (tf), currently, there doesn't seem to be a better alternative
+    # https://github.com/ros/geometry2/issues/32
+    # https://github.com/ros/robot_state_publisher/pull/30
+    # TODO(orduno) Substitute with `PushNodeRemapping`
+    #              https://github.com/ros2/launch_ros/issues/56
+    remappings = [('/tf', 'tf'),
+                  ('/tf_static', 'tf_static')]
+
+    # Declare the launch arguments
+    declare_use_rviz_cmd = DeclareLaunchArgument(
+        'use_rviz',
+        default_value='True',
+        description='Whether to start RVIZ')
+
+    declare_simulator_cmd = DeclareLaunchArgument(
+        'headless',
+        default_value='False',
+        description='Whether to execute gzclient)')
+
+    # Launch gazebo environment
+    start_gazebo_server_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution([
+                FindPackageShare('ros_gz_sim'),
+                'launch',
+                'gz_sim.launch.py'
+            ])
+        ),
+        launch_arguments={'gz_args': ['-r -s ' + world]}.items(),
+    )
+
+    start_gazebo_client_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            PathJoinSubstitution([
+                FindPackageShare('ros_gz_sim'),
+                'launch',
+                'gz_sim.launch.py'
+            ])
+        ),
+        launch_arguments={'gz_args': ['-g ' + world]}.items(),
+        condition=IfCondition(PythonExpression(['not ', headless])),
+    )
+
+    urdf = os.path.join(python_commander_dir, 'gz_tb3_waffle.urdf')
+
+    doc = xacro.parse(open(urdf))
+    xacro.process_doc(doc)
+
+    start_robot_state_publisher_cmd = Node(
+        package='robot_state_publisher',
+        executable='robot_state_publisher',
+        name='robot_state_publisher',
+        output='screen',
+        parameters=[{
+            'use_sim_time': True,
+            'robot_description': doc.toxml()
+        }],
+        remappings=remappings
+    )
+
+    # start the visualization
+    rviz_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(
+            os.path.join(nav2_bringup_dir, 'launch', 'rviz_launch.py')),
+        condition=IfCondition(use_rviz),
+        launch_arguments={'namespace': '',
+                          'use_namespace': 'False',
+                          'use_sim_time': 'true'}.items())
+
+    os.environ['IGN_GAZEBO_RESOURCE_PATH'] = os.path.join(
+        get_package_share_directory('turtlebot3_gazebo'), 'models')
+    os.environ['IGN_GAZEBO_RESOURCE_PATH'] += ':' + os.path.join(
+        get_package_share_directory('aws_robomaker_small_warehouse_world'), 'models')
+    os.environ['IGN_GAZEBO_RESOURCE_PATH'] += ':' + os.path.join(
+        get_package_share_directory('turtlebot3_gazebo'), '..')
+
+    # Clock bridge
+    clock_bridge = Node(
```

Comment:
It just occurred to me that we abstracted out the gz tb3 robot simulation launch file to deal with the gz specifics / bridges and whatnot. Can't that be used in all these launch files as well if we send it the updated starting pose?

Reply:
I made the PRs independent, because the tb3 for Gazebo classic is defined in the world file inside this package. We are using a two versions of the tb3, which are a little bit different between them.

But I'm happy to use the new resources included in `nav2_bringup`.  What fo you prefer ?

 - Keep models independent from nav2_bringup 
 - Use the same models here and in nav2_bringup

Reply:
Same - unless there are key differences... which I don't believe there should be? I _think_ when I made this I may have made the lidar for the simple commander range much higher to deal with the larger nature of the space that would need to be kept. 

So I'd say I'd rather re-use as much as possible. Where there are changes where they're not the same, err on the side of liberal use of resources (e.g. longer ranges and such)